### ISSUE TITLE
Fatal Exception: java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.plugin.common.EventChannel$EventSink.error(java.lang.String, java.lang.String, java.lang.Object)' on a null object reference

### ISSUE THREAD

**Issue by a1111198** [RodrigoSMarques/flutter_branch_sdk#108]
I am implementing a referral system using Branch.io Deep Link now the integration is working well for Debug mode and release mode but when I download the app through the play store then I don't get the expected data. I meant by release mode is that I am running the app in release mode using the code editor.

1. I am attaching the response of release mode which is working fine also the crash analytics that I got for users who tried to get the app from the Play Store.
this is response:

`listenDynamicLinks - DeepLink Data: {$keywords: [urbanculture], $og_title: I recommend you use the Urban Culture app for hassle free beauty services., $publicly_indexable: true, ~creation_source: 2, $og_description: Download app using link, +click_timestamp: 1623934435, source: android, $identity_id: 933689753256479021, ~stage: new share, $og_image_url: https://firebasestorage.googleapis.com/v0/b/urbanculture5.appspot.com/o/appGraphics%2FWhatsApp%20Image%202021-05-27%20at%205.35.15%20PM.jpeg?alt=media&token=7076c9a4-4397-4e07-91d8-bff5392bed1b, ~feature: sharing, url: http://www.google.com, +match_guaranteed: true, user_id: xWhW37rO1XOBLYCZRevdCHxRfGI2, $canonical_identifier: Urbanculture, +clicked_branch_link: true, $one_time_use: false, ~id: 933690012177310442, +is_first_session: false, ~referring_link: https://urbanculture.app.link/a3oOkqot8gb, ~channel: unknown}`

What I noticed that sometimes I got this response but sometimes this issue: 
**Fatal Exception: java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.plugin.common.EventChannel$EventSink.error(java.lang.String, java.lang.String, java.lang.Object)' on a null object reference**

I am using in **Flutter flutter_branch_sdk: ^1.4.0** 
3. Code For Link generation:
  ```
BranchUniversalObject buo = BranchUniversalObject(
      canonicalIdentifier: '********',
      title:
          'I recommend you use the ******* app for hassle-free beauty services.',
      imageUrl:
          'https://firebasestorage.googleapis.com/v0/b/***********.com/o/appGraphics%2FWhatsApp%20Image%202021-05-27%20at%205.35.15%20PM.jpeg?alt=media&token=7076c9a4-4397-4e07-91d8-bff5392bed1b',
      contentDescription: 'Download app using link',
      keywords: [********],
      publiclyIndex: true,
      locallyIndex: true,
      contentMetadata: BranchContentMetaData()
        ..addCustomMetadata(
          'user_id',
          FirebaseAuth.instance.currentUser != null
              ? '${FirebaseAuth.instance.currentUser.uid}'
              : 'null',
        ),
    );
    BranchLinkProperties lp = BranchLinkProperties(
        channel: 'unknown', feature: 'sharing', stage: 'new share', tags: []);
    lp.addControlParam('url', 'http://www.google.com');
    shortURl = await FlutterBranchSdk.getShortUrl(buo: buo, linkProperties: lp);
    if (shortURl.success) {
      print('Link generated: ${shortURl.result}');
      setState(() {
        shortURl = shortURl;
      });
    } else {
      //  print('Error : ${response.errorCode} - ${response.errorMessage}');
    } ```
**and Depp link receiving code is :**

   ```streamSubscription = FlutterBranchSdk.initSession().listen((data) {
      print("DDDD");
      print('listenDynamicLinks - DeepLink Data: $data');
      controllerData.sink.add((data.toString()));
      if (data.containsKey('+clicked_branch_link') &&
          data['+clicked_branch_link'] == true) {
        print('Custom string: ${data['user_id']}');
        refreeId = data['user_id'];
      }
    }, onError: (error) {
      PlatformException platformException = error as PlatformException;
      print(
          'InitSession error: ${platformException.code} - ${platformException.message}');
    });
```

Also attached screenshots of Crashlytics.

![crash1](https://user-images.githubusercontent.com/60999356/122567807-b5301180-d066-11eb-9fe6-24f35030cc56.jpg)
![crash2](https://user-images.githubusercontent.com/60999356/122567810-b5c8a800-d066-11eb-802d-4332be282dce.jpg)
![crash3](https://user-images.githubusercontent.com/60999356/122567803-b3664e00-d066-11eb-8e97-77606128f7e0.jpg)

**Comment by RodrigoSMarques** [comment]
@a1111198 
This bug actually exists in version 1.4 and was fixed in version 3.2 (null safety).

**Comment by a1111198** [comment]
I have not enabled null safety in my project and not upgraded to 2.0. So any possible solution around it in this version (means some slight modification in 1.4 SDK)?

**Comment by RodrigoSMarques** [comment]
I will evaluate. I can generate a version, but it won't be published to pub.dev.

You would use the package with reference to the GIT.

**Comment by a1111198** [comment]
Thanks, I am waiting for it.

**Comment by RodrigoSMarques** [comment]
I made the correction. Update your `pubspec.yaml` to

```dart
  flutter_branch_sdk:
    git:
      url: https://github.com/RodrigoSMarques/flutter_branch_sdk.git
      ref: v1.4.1
```

Test and report the result.

**Comment by a1111198** [comment]
I really Thank you for making your efforts. but I am getting this error in compiling:

```
C:\Users\ASUS\flutter\.pub-cache\git\flutter_branch_sdk-f06253209579146d66c86328c3e6da494aebff9f\android\src\main\java\br\com\rsmarques\flutter_branch_sdk\FlutterBranchSdkPlugin.java:56: error: package StrieventSink does not exist
    private Map<StrieventSink.errorng, Object> initialParams = null;
                             ^
Note: C:\Users\ASUS\flutter\.pub-cache\git\flutter_branch_sdk-f06253209579146d66c86328c3e6da494aebff9f\android\src\main\java\br\com\rsmarques\flutter_branch_sdk\FlutterBranchSdkPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
```
I have used flutter clean. and I am not using flutter 2.0 and not enabled null Safety. I am really thankful for showing this much interest in this issue. I hope you will fix it very soon.

**Comment by RodrigoSMarques** [comment]
Hi.
In sleep I ended up changing a snippet of code without seeing.

I made the correction. Update your `pubspec.yaml` to

```dart
  flutter_branch_sdk:
    git:
      url: https://github.com/RodrigoSMarques/flutter_branch_sdk.git
      ref: v1.4.1+1
```

Test and report the result.

**Comment by RodrigoSMarques** [comment]
@a1111198 

Can I close the issue?

**Comment by a1111198** [comment]
I tried it and I didn't get this error but what I figured is that it is not working in release mode. I don't know what happened. Then I got busy somewhere so couldn't figure out the actual cause. I will check it again in a day or two and update here what is the issue if not found issue then I will close it.

**Comment by RodrigoSMarques** [comment]
Try deleting the cache folder in the path: C:\Users\ASUS\flutter\.pub-cache\git\

I await your feedback.

**Comment by a1111198** [comment]
I cleared the cache as you mentioned in the above comment. I am not getting this error this time, but this is working well for the released APK (using ?bnc_validate=true), but when downloading the app from the play store, I am not getting associated data to the link. I am not getting any crash in my Crashanalytics so can't mention that here. 
can you please confirm which SHA256 key should I use in branch config? Should I use SHA256 of the key from which I signed my App or use the play store signing key?   
Now you can close this issue because this issue is not occurring but my problem is not solved yet. what should I do?

**Comment by RodrigoSMarques** [comment]
Hi @a1111198 
Instructions for key here: [https://help.branch.io/developers-hub/docs/android-troubleshooting#generate-signing-certificate](https://help.branch.io/developers-hub/docs/android-troubleshooting#generate-signing-certificate)

Do you have active proguard?
Please check the instructions here.[https://help.branch.io/developers-hub/docs/android-full-reference#proguard-settings-for-leveraging-branchs-pooled-matching](https://help.branch.io/developers-hub/docs/android-full-reference#proguard-settings-for-leveraging-branchs-pooled-matching)

**Comment by a1111198** [comment]
regrading proguard this is what I am using :
```
#Flutter Wrapper
-keep class io.flutter.app.** { *; }
-keep class io.flutter.plugin.**  { *; }
-keep class io.flutter.util.**  { *; }
-keep class io.flutter.view.**  { *; }
-keep class io.flutter.**  { *; }
-keep class io.flutter.plugins.**  { *; }
-keep class com.facebook.applinks.** { *; }
-keepclassmembers class com.facebook.applinks.** { *; }
-keep class com.facebook.FacebookSdk { *; }
-keep class com.google.android.gms.** { *; }
-keep class com.huawei.hms.ads.** { *; }
-keep interface com.huawei.hms.ads.** { *; }
```
As mentioned in the document that Key must be obtained from this: `keytool -list -v -keystore my-release-key.keystore `is same as key.jks file in android/app/key.kjs

**Comment by slorop41314** [comment]
Also got this issue in release mode, working fine on debug mode.

Flutter null safety
flutter_branch_sdk: ^3.2.0
  
<img width="1048" alt="Screen Shot 2021-07-13 at 10 41 04" src="https://user-images.githubusercontent.com/52713279/125386649-d4e8fa00-e3c6-11eb-80ac-f381d2b7aed6.png">

**Comment by RodrigoSMarques** [comment]
> Also got this issue in release mode, working fine on debug mode.
> 
> Flutter null safety
> flutter_branch_sdk: ^3.2.0
> 
> <img alt="Screen Shot 2021-07-13 at 10 41 04" width="1048" src="https://user-images.githubusercontent.com/52713279/125386649-d4e8fa00-e3c6-11eb-80ac-f381d2b7aed6.png">


Strange. The problem has fixed in version 3.2.
Please, expand the stack from the first line of the print e send for me.

**Comment by slorop41314** [comment]
Sorry my bad, seems like its already fixed in 3.2. Thanks.

**Comment by RodrigoSMarques** [comment]
Closed. No activity in the last 14 days.
If necessary open again.
